### PR TITLE
Add test coverage support to android_local_test

### PR DESCRIPTION
**Pull request by Bencodes**
Adding test coverage support to `android_local_test`.

https://github.com/bazelbuild/bazel/issues/15827

### CODE REVIEW

**Review by Tagakov — APPROVED** [review]

**Review by ted-xie — CHANGES_REQUESTED** [review]

File: src/test/shell/bazel/android/android_local_test_integration_test.sh [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  public void testBar() {
+    new Bar();
+  }
+}
+EOF
+}
+
+# Asserts that the coverage file exists by looking at the current $TEST_log.
+# The method fails if TEST_log does not contain any coverage report for a passed test.
+function assert_coverage_file_exists() {
```
[the remark this answers is not in the document]
Comment (ted-xie) [reply src/test/shell/bazel/android/android_local_test_integration_test.sh]: It looks like there were some Windows-related failures in the latest presubmit; any idea about those?

[the remark this answers is not in the document]
Comment (Bencodes) [reply src/test/shell/bazel/android/android_local_test_integration_test.sh]: @ted-xie everything should be passing. The windows failures were build flakes.

**Comment by sofiethorsen** [comment]
@ted-xie, @lberki, @ahumesky - are there any outstanding remarks you have on this PR that would block merging? I'm currently excluding all android tests when running coverage (only collecting coverage for pure Kotlin) - it would be beneficial to be able to include the android ones too.

**Comment by ted-xie** [comment]
@sofiethorsen thanks for pinging me. I think we should merge this in.

**Review by ted-xie — APPROVED** [review]

**Comment by keertk** [comment]
@bazel-io fork 6.1.0

**Pull request closed without merging**